### PR TITLE
Add socket enum classes from py3.4+

### PR DESCRIPTION
**Pull request by jreese**
Adds four IntEnum classes in the socket module that mirror the
AF_, AI_, MSG_, and SOCK_ sets of constants.

### CODE REVIEW

**Comment by jreese** [comment]
Running `python tests/mypy_test.py` on my mac with Py3.6.1 failed with the following errors, even from a fresh clone:  https://gist.github.com/jreese/d95da7c166a81c8161b02c4f3bff98d8

**Comment by gvanrossum** [comment]
What version of mypy did you have installed for that test?

**Comment by jreese** [comment]
I used `pip install -r requirements-tests-py3.txt` in a 3.6.1 virtualenv following the readme:

    (.venv3) jreese@jreese-mbp ~/workspace/typeshed +socket-enums⚡ » pip list
    attrs (16.3.0)
    flake8 (3.3.0)
    flake8-bugbear (17.3.0)
    flake8-pyi (17.3.0)
    mccabe (0.6.1)
    mypy (0.501)
    pip (9.0.1)
    pycodestyle (2.3.1)
    pyflakes (1.5.0)
    setuptools (28.8.0)
    typed-ast (1.0.2)

**Comment by gvanrossum** [comment]
Can you try again with mypy master?

**Comment by jreese** [comment]
Yep, that resolved the errors!

File: stdlib/3/socket.pyi [hunk 1]
```diff
@@ -250,6 +252,68 @@ TIPC_WITHDRAWN = 0
 TIPC_ZONE_SCOPE = 0
 
 
+# enum versions of above flags py 3.4+
+if sys.version_info >= (3, 4):
+    class AddressInfo(IntEnum):
```
Comment (JelleZijlstra) [inline stdlib/3/socket.pyi]: This one and MsgFlag actually inherit from IntFlags.

File: stdlib/3/socket.pyi [hunk 2]
```diff
@@ -250,6 +252,68 @@ TIPC_WITHDRAWN = 0
...
+        AI_ADDRCONFIG = ...
```
Comment (JelleZijlstra) [inline stdlib/3/socket.pyi]: Confirmed locally that you don't need types for the values, at least with mypy. (So no action required from you.)

**Comment by gvanrossum** [comment]
Please wait with merging until I've tested this against our internal codebase.

File: stdlib/3/socket.pyi (hunk 1 above)
Comment (jreese) [inline stdlib/3/socket.pyi]: Good point, I'll update the pull request.

**Comment by jreese** [comment]
Is there any progress on testing?  I updated the PR with fixes for @JelleZijlstra's comments and only include `AddressInfo` and `MsgFlag` for 3.6+.

**Comment by gvanrossum** [comment]
The tests with internal codebase pass, so +1 from me. I'll let Jelle decide whether he's happy with the rest.

**Comment by JelleZijlstra** [comment]
I'll take another look within a few days (tonight if I have time) and then merge.

**Pull request merged**